dali/imgcodec/image_decoder_test.cc

```diff
@@ -301,16 +434,65 @@ class ImageDecoderTest : public ::testing::Test {
     return ctx;
   }
 
+  void FilterDecoder(std::function<bool(ImageDecoderFactory *)> decoder_filter) {
+    this->SetDecoder(std::make_unique<ImageDecoder>(this->GetDeviceId(), false,
+                                                    std::map<std::string, any>{}, decoder_filter));
+  }
+
   void DisableFallback() {
     // making sure that we don't pick the fallback implementation
     auto filter = [](ImageDecoderFactory *factory) {
-      if (dynamic_cast<OpenCVDecoderFactory *>(factory) != nullptr) {
-        return false;
-      }
-      return true;
+      return dynamic_cast<OpenCVDecoderFactory *>(factory) == nullptr;
     };
-    this->SetDecoder(std::make_unique<ImageDecoder>(this->GetDeviceId(), false,
-                                                    std::map<std::string, any>{}, filter));
+    FilterDecoder(filter);
+  }
+
+
+  float GetEps() {
+    float eps = 0.01f;
+    if (!std::is_floating_point_v<OutputType>) {
+      // Adjusting the epsilon to OutputType
+      eps *= max_value<OutputType>();
+    }
+    return eps;
+  }
+
+  void CompareData(const TensorView<StorageCPU, const OutputType> &data,
+                   const test_sample &sample) {
+    if constexpr (std::is_same<Backend, GPUBackend>::value) {
```

Comment:
Well.... we should use it on a per-sample basis - only for stuff we know may differ (i.e. NVJPEG) - with this you allow big errors in decoders that really run on the CPU and the result is just being copied. I guess that unless we use mislabeled data (and I don't think we do), this could be something like

```suggestion
    if (std::is_same<Backend, GPUBackend>::value && sample.filename.find(".jpg") != string::npos) {
```